### DISCUSSION TITLE
출석부 수정 API 추가에 따른 출석체크 API 수정 의견

### DISCUSSION THREAD

**Discussion by echo724** [i-am-here-too/iht-backend#6] [개선사항 의견]
### 문맥
- 출석부 자체에 대한 날짜 변경이 가능하도록 수정 API를 추가해야하는 요구사항 발생 https://github.com/haru-attendance/haru-backend/pull/4#discussion_r1472681262
- 따라서 현재 출석체크 API(`PUT /clubs/<club_id>/attendance/<attendace_id>`)가 출석부 수정 API로 변경되어야한다.
- 새로운 출석체크 API가 필요한 상황

#### 1. attendance와 member id로 출석체크 - `PATCH or PUT  /clubs/<club_id>/attendances/<attendace_id>/members/<member_id>`
이유: 도메인 주도 개발에서 각각의 에그리거트를 에그리거트 루트를 통해 접근하듯이, 서버에 요청을 보낼때도 각각의  attendance와 member 에그리거트 루트 id들을 통해 출석부-구성원의 출석현황(attended) 필드 수정

#### 2. attendance-member id로 직접 출석체크 - `PATCH or PUT /clubs/<club_id>/attendance-members/<attendance-member-id>`
이유: 어차피 출석부 조회할 때에 attendance-member id를 직접 내려주기 때문에 해당 id를 알 수 있고 바로 접근 가능

### 참고
[REST API 설계 원칙](https://sharplee7.tistory.com/49)

참고 블로그 내용 한 번 보면 REST API를 이해하기 쉽더라구요. 시간 되면 읽어보시고 어떤 방식이 좋을지 의견 공유해주시면 감사하겠습니다!

**Comment by kong-hana01** [comment]
잘 정리해주셔서 감사합니다 에코!

사실 restful 하게 api를 구성하는 거라고 하면 1번이 더 적절해보이기는 합니다!
그런데 `url로 attendance나 member의 id가 노출`되는 것이 신경쓰이네요!
전에 에코가 말한 것처럼 외부로 노출될 필요가 없는 정보가 노출되는 점이 위험해보여요.
그래서 `외부에 정보를 노출시키지도 않으면서 조회 성능도 더 좋을 것으로 예상되는 2번`이 좋아보입니다!

다른 분들께서는 어떻게 생각하시나요?

**Reply by echo724** [reply]
좋습니다~ 다만 club_id도 url에 필요한지는 여전히 고민이긴 하네요!
